src/shared/functions.ts

```diff
@@ -81,3 +90,46 @@ export function getDemographicLabel(id: string) {
     ? "Pacific Islander"
     : id.split(/(?=[A-Z])/).join(" ");
 }
+
+export const getMetricFieldForDemographicsId = (id: string) =>
+  id === "population" ? id : `${id}Population`;
+
+export function getDemographicsMetricFields(staticMetadata: IStaticMetadata): MetricsList {
+  // eslint-disable-next-line functional/prefer-readonly-type
+  const data: (readonly [string, string])[] = staticMetadata.demographics.flatMap(file =>
+    CORE_METRIC_FIELDS.includes(file.id)
+      ? []
+      : [[file.id, getMetricFieldForDemographicsId(file.id)]]
+  );
+  // Need to loosen up the types here
+  const order: readonly string[] = DEMOGRAPHIC_FIELDS_ORDER;
+  // eslint-disable-next-line functional/immutable-data
+  data.sort(([, a], [, b]) => order.indexOf(a) - order.indexOf(b));
```

Comment:
```suggestion
  data.sort(([a,], [b,]) => order.indexOf(a) - order.indexOf(b));
```
The first item is the key, not the second. Since the field names aren't in the `order` list, sorting by that leaves the order unchanged.